### ISSUE TITLE
On the issue of fat gain from milk

### ISSUE THREAD

**Issue by SilearFlare** [Hirachosan/FattyFoodsAddon#2]
Hello, sorry to bother you again! 
I'm really liking the mod. This time it's not a technical issue but a more in-depth question about the game mechanics, specifically how fat gain from milk seems to be breaking the game. 

Milk is, realistically, incredibly nutrient rich. A bucket can give 1 shank and almost a full shank of fat. What that means is that basically, as soon as you can get your hands on a bucket, you can milk random cow herds with full udders, to get around 4 shanks of food AND fat, almost filling up your fat bar halfway through. This only becomes worse when you get yourself a pair of shears and build a cow pen out of dirt and grass slabs to spawnproof it, which, again, is very earlygame stuff.
![2021-12-20_20 08 17](https://user-images.githubusercontent.com/5557920/147077376-47e026de-39ee-4ad4-833f-663313518469.png)

This is my cow pen. It requires no input, I just swing by in the morning and drink a couple of buckets. Even with hardcore starvation enabled, just being able to drink those two buckets of milk a day for its food content has been severely impacting my fat bar over the course of a week. I've tried to use it to my advantage by using a hand crank, but fat gets only burned at 1/4th of the speed of actual hunger bars, and it's just not working. This is a direct and extreme, but very telling example of how it works in a creative world. I fill myself up with milk until obese, full hunger and fat bar, and then start cranking. I lose about a shank and a half before becoming too hungry to continue.
![2021-12-22_10 57 28](https://user-images.githubusercontent.com/5557920/147077994-237467db-eaa8-4b61-8830-a0668a47206f.png)
I am now fat and hungry. I will need to eat more, and thus gain back the fat I lost, continuously becoming fatter in the process. The problem is something that doesn't present itself at all in vanilla BTW, since milk doesn't give you any fat. On top of that the fact that you can starve to death with any kind of fat content, which is a vanilla mechanic, makes milk an extremely punishing earlygame food compared to vanilla BTW, given the loss of sprint by just filling up your fat bar in the first place, which is not the case in the core game at all.
![2021-12-22_11 00 18](https://user-images.githubusercontent.com/5557920/147079103-549313ef-599a-4a95-8cd5-af9c7408093c.png)

My question is: is any of this intended to make sure people don't abuse cows in the earlygame? If not, given the circumstances, does this look like it could be a balance problem for the mod? Would removing or highly nerfing fat gains from plain milk fix this, do you think?

Thank you again for your time, I'm sorry if this isn't the FFA thread on the BTW forums, I just prefer making these posts here instead of going through the motions of making an account, verifying it, logging in and posting compared to the literal drag and drop of just doing this on github during my daily repo checks.

**Comment by Hirachosan** [comment]
Thanks for the feedback!  This is exactly how it was intended, although it might be tweaked to a be a little less severe. The latest patch overhauled quite a bit and i might have not reconsider milk again.. 

Im totally fine with you writing your thoughts here, if anything its more organized :) maybe less people will read it though.

Now about the question: it is intended yes, but not for the reason you state per say. The main concept for FFA is to use fat to gain variety. Fat Isn't meant to be beneficial completely. The same goes for pork btw, eating pork alone would mean you gain more fat than you can lose :) so food wise ideally you want to mix different kinds of food, have some fat for that slow burning extra hunger!

As you get more stable in the game fat becomes more of a nuisance and limiter. It wont be hard to gain fat at all(to be fair, no one really keeps cows until perma light so i consider getting milk on a daily basis fairly advanced) 
Why is it a limiter? Well.. check out the desserts category! You can make early game desserts like pure sugar (sharp stone plus reed craft, yes you can directly eat sugar now) and cocoa beans give you some special buffs.. later game dessert have better and more unique buffs, but you can't spam those because you'll get fat :p 

TL;DR: fat isn't just an extra hunger bar, it gives pros and cons and there's a lot of complex interactions when you fully use the system, it is indeed intended that some foods give more fat than you'd lose:)

**Comment by Hirachosan** [comment]
Just imagine, in vanilla BTW, there's no wat you could end up chubby without intending it :D

**Comment by Hirachosan** [comment]
The hand crank pit is exactly what i used to try and balance foods! Here's something else i thought of.. hand cranks are basically late game cardio machines.. when you want more food buffs, you pul out some hand cranks and start eating low fat food to lose the rolls you want to lose :D so you can buff yourself some more at the cost of extra hunger

**Comment by SilearFlare** [comment]
Ah, I think I finally get it now. Thank you for the in-depth explanation! I've managed to get around with early cows with the new grass slabs blocks, which stops mob spawns in the pen while also at the same time doubling as a food source for cows, and requiring a fairly large pen for just two heads of cattle to survive.

**Comment by Hirachosan** [comment]
Grass slabs don't stop mob spawns :p in BTW mobs can spawn anywhere they fit!  So even on slabs. The only thing that stops spawns is wood/wicker(plant based i guess) and being close to the player, i think your cowpen has been safe so far cause its really close to your living area?

**Comment by SilearFlare** [comment]
Huh. Damn, that's news, I guess I just got really lucky
